cli/src/config.ts

```diff
@@ -1,354 +1,156 @@
+      androidStudioPath:
+        'C:\\Program Files\\Android\\Android Studio\\bin\\studio64.exe',
+    },
+    linux: {
+      androidStudioPath: '/usr/local/android-studio/bin/studio.sh',
+    },
+    android: await loadAndroidConfig(appRootDir, cli.assetsDir),
+    ios: await loadIOSConfig(appRootDir, cli.assetsDir),
+    web: {
+      name: 'web',
+    },
+    cli,
+    app: {
+      rootDir: appRootDir,
+      appId,
+      appName,
+      webDir,
+      webDirAbs: resolve(appRootDir, webDir),
+      package: (await readPackageJSON(resolve(appRootDir, 'package.json'))) ?? {
+        name: appName,
+        version: '1.0.0',
+      },
+      extConfigName: EXTERNAL_CONFIG_FILE,
+      extConfigFilePath: resolve(appRootDir, EXTERNAL_CONFIG_FILE),
+      extConfig,
+      bundledWebRuntime: extConfig.bundledWebRuntime ?? false,
+    },
+  };
+}
 
-let Package: PackageJson;
-let ExtConfig: ExternalConfig;
+async function loadCLIConfig(rootDir: string): Promise<CLIConfig> {
+  const assetsName = 'assets';
 
-export class Config implements CliConfig {
-  windows = {
-    androidStudioPath:
-      'C:\\Program Files\\Android\\Android Studio\\bin\\studio64.exe',
+  return {
+    rootDir,
+    assetsName,
+    assetsDir: join(rootDir, assetsName),
+    package: await readJSON(resolve(rootDir, 'package.json')),
+    os: determineOS(process.platform),
   };
+}
 
-  linux = {
-    androidStudioPath: '/usr/local/android-studio/bin/studio.sh',
-  };
+async function loadAndroidConfig(
+  rootDir: string,
+  assetDir: string,
+): Promise<AndroidConfig> {
+  const name = 'android';
+  const platformDir = resolve(rootDir, name);
+  const webDir = 'app/src/main/assets/public';
+  const resDir = 'app/src/main/res';
+
+  const templateName = 'android-template';
+  const pluginsFolderName = 'capacitor-cordova-android-plugins';
 
-  android = {
-    name: 'android',
+  return {
+    name,
     minVersion: '21',
-    platformDir: '',
-    webDir: 'app/src/main/assets/public',
-    webDirAbs: '',
-    resDir: 'app/src/main/res',
-    resDirAbs: '',
+    platformDir,
+    webDir,
+    webDirAbs: resolve(platformDir, webDir),
+    resDir,
+    resDirAbs: resolve(platformDir, resDir),
     assets: {
-      templateName: 'android-template',
-      pluginsFolderName: 'capacitor-cordova-android-plugins',
-      templateDir: '',
-      pluginsDir: '',
+      templateName,
+      pluginsFolderName,
+      templateDir: resolve(assetDir, templateName),
+      pluginsDir: resolve(assetDir, pluginsFolderName),
     },
   };
+}
 
-  ios = {
-    name: 'ios',
+async function loadIOSConfig(
+  rootDir: string,
+  assetDir: string,
+): Promise<IOSConfig> {
+  const name = 'ios';
+  const platformDir = resolve(rootDir, name);
+  const webDir = 'public';
+  const nativeProjectName = 'App';
+  const templateName = 'ios-template';
+  const pluginsFolderName = 'capacitor-cordova-ios-plugins';
+
+  return {
+    name,
     minVersion: '11.0',
     cordovaSwiftVersion: '5.0',
```

Comment:
```suggestion
    cordovaSwiftVersion: '5.1',
```
We updated the Capacitor version to 5.1, so this one should be probably updated too